### ISSUE TITLE
active-sessions-limit-handler is not working when email as username is enabled

### ISSUE THREAD

**Issue by rksk** [wso2/product-is#13121]
**Describe the issue:**
With the fix for https://github.com/wso2/product-is/issues/12169, the following logic is added to `storeSessionData()` of `DefaultAuthenticationRequestHandler`.
```
// If the user is federated, generate a unique ID for the user and add an entry to the IDN_AUTH_USER
// table with the tenant id as -1 and user store domain as FEDERATED.
if (LOCAL_IDP_NAME.equals(idpName) && !containIllegalCharacters(userName)) {
   userId = FrameworkUtils.resolveUserIdFromUsername(tenantId, userStoreDomain, userName);
} else {
```

```
    /**
     * Checks whether the given string contains "/" or "@"
     *
     * @param str String to be checked.
     * @return True if contains above characters, False otherwise.
     */
    private boolean containIllegalCharacters(String str) {

        return (str.contains(UserCoreConstants.DOMAIN_SEPARATOR) ||
                str.contains(UserCoreConstants.TENANT_DOMAIN_COMBINER));
    }
```

Now, if email username is enabled, the username is considered as `containIllegalCharacters() == true` and the user id will be picked from the UserSessionStore while it should pick the user id from `FrameworkUtils.resolveUserIdFromUsername()` for any local user.

Therefore, the session mapping is stored against the wrong user id and it will not be fetched correctly from the active-sessions-limit-handler.

**How to reproduce:**
- Enable [Email Address as the Username](https://is.docs.wso2.com/en/latest/learn/using-email-address-as-the-username/#using-email-address-as-the-username)
- Configure [active-sessions-limit-handler](https://is.docs.wso2.com/en/latest/learn/limiting-active-user-sessions-based-on-criteria/#limiting-active-user-sessions-based-on-criteria)
- But it will not prompt to revoke the sessions for any users

**Environment information**
 - Product Version: IS 5.10.0 (an update level after Aug 2021)

**Comment by ruwanta** [comment]
It is unclear the reasons for
1. store tenant ID as -1
2. Why "@" sign is considered unacceptable character

**Comment by piraveena** [comment]
@Yasara-Y do we have anything pending on this?

**Comment by Yasara-Y** [comment]
Yes, the public fix of this issue is rather complex since it involves multiple fixes on top of a currently ongoing API session management feature currently being implemented to the public master tracked with [1]. Once that feature is completed, I will start working on the public fix.

[1] https://github.com/wso2/product-is/issues/12193

**Comment by Yasara-Y** [comment]
@ThaminduDilshan is working on the public fix currently.

**Comment by ThaminduDilshan** [comment]
This is a regression caused by the original fix. We are doing the fix in a different manner for the public branch.
Hence not reproducible with https://github.com/wso2/carbon-identity-framework/pull/4185 or https://github.com/wso2/identity-api-user/pull/155 (yet to merge)

However the bug still persists. Hence opening a [new issue](https://github.com/wso2/product-is/issues/14542) for that
